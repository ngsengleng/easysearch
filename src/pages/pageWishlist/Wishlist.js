import React, { useEffect, useState } from "react";
import RenderResults from "../../components/RenderResults";
import "firebase/database";
import "firebase/firestore";
import { firebase } from "@firebase/app";
import { Typography, makeStyles, Link } from "@material-ui/core";
import { useHistory } from "react-router-dom";
import GeneralHeader from "../../components/GeneralHeader";

const db = firebase.firestore();

const useStyles = makeStyles({
  pageHeader: {
    margin: "auto",
    textAlign: "center",
    marginTop: "10px",
  },

  homeLink: {
    color: "black",
    margin: "auto",
    textAlign: "center",
    marginTop: "10px",
  },
});

export default function Wishlist() {
  const history = useHistory();
  const classes = useStyles();
  const [wishlist, setWishlist] = useState([]);
  const [bool, setBool] = useState();
  const goToHome = () => history.push("/");
  // fetch data from firestore on page render
  useEffect(() => {
    document.title = "Wishlist";
    let unsubscribe = "";
    const fetchWishlist = async () => {
      const currentUser = firebase.auth().currentUser.uid;
      const data = await db
        .collection("users")
        .doc(currentUser)
        .collection("wishlist")
        .doc("arrayOfItems");
      unsubscribe = data.onSnapshot((doc) => {
        if (doc.exists) {
          setWishlist(doc.data().items);
          setBool(true);
        } else {
          console.log("no such document");
        }
      });
    };
    fetchWishlist();
    return () => {
      unsubscribe();
    };
  }, []);
  return (
    <div>
      <Typography
        variant="h3"
        display="block"
        gutterBottom
        className={classes.pageHeader}
      >
        Wishlist
      </Typography>
      <GeneralHeader />
      {wishlist === undefined || wishlist.length === 0 ? (
        <div>
          <Typography
            variant="h5"
            display="block"
            gutterBottom
            className={classes.pageHeader}
          >
            Your wishlist is empty
          </Typography>
          <Typography variant="subtitle1" className={classes.homeLink}>
            <Link href="/" onClick={goToHome} color="inherit">
              Search for some items now!
            </Link>
          </Typography>
        </div>
      ) : (
        wishlist?.map((entry, idx) => {
          return (
            <RenderResults
              key={idx}
              itemData={entry}
              bool={bool}
              store={entry.store}
            />
          );
        })
      )}
    </div>
  );
}
