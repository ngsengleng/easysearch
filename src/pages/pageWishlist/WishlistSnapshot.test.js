import React from "react";
import renderer from "react-test-renderer";
import Wishlist from "./Wishlist";
import { AuthProvider } from "../../context/AuthContext";
it("renders correctly", () => {
  const tree = renderer
    .create(
      <AuthProvider>
        <Wishlist />
      </AuthProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
