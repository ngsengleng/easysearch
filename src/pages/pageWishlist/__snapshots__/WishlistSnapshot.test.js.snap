// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-pageHeader-1"
  >
    <h3
      className="MuiTypography-root makeStyles-pageHeader-1 MuiTypography-h3 MuiTypography-gutterBottom MuiTypography-displayBlock"
    >
      Wishlist
    </h3>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Subcribe to notifications now!
      </span>
    </button>
  </div>
  <div>
    <div
      className="MuiGrid-root makeStyles-resultHeader-6 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          IMAGE
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          ITEM NAME
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          price
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          store
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          RATING
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          WISHLIST
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          className="MuiTypography-root MuiTypography-button MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          link to site
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root makeStyles-divider-5"
    />
  </div>
  <div>
    <h5
      className="MuiTypography-root makeStyles-pageHeader-1 MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-displayBlock"
    >
      Your wishlist is empty
    </h5>
    <h6
      className="MuiTypography-root makeStyles-homeLink-2 MuiTypography-subtitle1"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Search for some items now!
      </a>
    </h6>
  </div>
</div>
`;
